visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -112,6 +119,17 @@ public VisualApi build() {
   private FullPageScreenshotConfig fullPageScreenshotConfig;
   private String sessionMetadataBlob;
 
+  /**
+   * Creates a VisualApi instance for a given Visual Backend {@link DataCenter}
+   *
+   * @param driver The {@link org.openqa.selenium.WebDriver} instance where the tests should run at
+   * @param username SauceLabs username
+   * @param accessKey SauceLabs access key
+   */
+  public VisualApi(RemoteWebDriver driver, String username, String accessKey) {
+    this(driver, "", username, accessKey);
```

Comment:
Wouldn't it be better if this is resolved prior invoking the other constructor ?

Reply:
I agree that this doesn't read great, however this implementation is due to a limitation of java regarding usage of `this` keyword. It has be the first thing executed when used in a constructor and no other function calls are allowed inside the statement. To modify this behaviour we need to rework the constructor calling logic, which is probably not something we want to do for this task

Reply:
Turns out that static functions are indeed callable using `this` in constructor. 🤦 
It should look better now.